### PR TITLE
docs(markdown-magic): readme files generated with markdown-magic

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Summary:

This change implements [Markdown Magic](https://www.npmjs.com/package/markdown-magic) and updates both `readme.md` files to use new import files for each topic.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,14 +38,16 @@
...
     "npm-run-all": "^4.1.5",
     "semantic-release": "^17.4.7"
   },
   "scripts": {
     "build:test": "node bin/generate.js -t -n Han-Solo -P @mypackage -d ./auro-test",
     "build:complete": "node bin/generate.js -n auro-solo -d ../auro-test",
     "sweep": "rm -rf auro-test",
-    "test": "npm run build:test"
+    "test": "npm run build:test",
+    "docs": "node util/generateDocs.js"
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply package.json:50]: Applied change manually to suggested script name

File: docs/topics/usage/developmentApi.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+| Command | Description |
+| --- | --- |
+| `build:test` | Will test the generate pipeline w/o the install process at `./auro-test`
+| `build:complete` | Will build a new custom element with complete install process at `../auro-test`
+| `sweep` | Will delete auro-test directory
+| `test` | Runs `build:test`
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply docs/topics/usage/developmentApi.md:6]: Fixed manually

File: README.md [hunk 3]
```diff
@@ -1,71 +1,136 @@
...
+<!-- AURO-GENERATED-CONTENT:START (FILE:src=./docs/topics/badges.md) -->
+<!-- The below content is automatically added from ./docs/topics/badges.md -->
+[![Build Status](https://img.shields.io/github/workflow/status/AlaskaAirlines/WC-Generator/Test%20and%20publish?branch=master&style=for-the-badge)](https://github.com/AlaskaAirlines/WC-Generator/actions?query=workflow%3A%22test+and+publish%22)
+[![See it on NPM!](https://img.shields.io/npm/v/@aurodesignsystem/wc-generator.svg?style=for-the-badge&color=orange)](https://www.npmjs.com/package/@aurodesignsystem/wc-generator)
+[![License](https://img.shields.io/npm/l/@aurodesignsystem/wc-generator.svg?color=blue&style=for-the-badge)](https://www.apache.org/licenses/LICENSE-2.0)
+[![issues](https://img.shields.io/github/issues-raw/AlaskaAirlines/WC-Generator?style=for-the-badge)](https://github.com/AlaskaAirlines/WC-Generator/issues)
+<!-- AURO-GENERATED-CONTENT:END -->
+
+<!-- AURO-GENERATED-CONTENT:START (TOC) -->
+- [Description](#description)
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply README.md:12]: @blackfalcon Doing something like this with the existing platform shouldn't be too hard. What do you think about implementing this for now until we do a style makeover on the doc site?

Screen Shot 2021-11-04 at 7 15 46 PM

**Comment by jason-capsule42** [comment]
> This commit shouldn't be a feature release. Yes, it is adding a _feature_ to the repo, but it's not a feature added to the generator itself from a consumer's perspective.
> 
> I'd suggest updating the commit to be `docs(markdown-magic):...` instead.

Done

File: template/docs/topics/development.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# Development
```
Comment (blackfalcon) [inline template/docs/topics/development.md:1]: Why is this file created? It's not being exposed in the repo as a README and is not included in any other file.

File: template/docs/topics/install.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+1. Define the dependency within each component that is using `<[namespace]-[name]>`.
+   ```javascript
+   import "[npm]/[namespace]-[name]";
+   ```
+
+1. Reference `[namespace]-[name]` in HTML.
+   ```html
+   <[namespace]-[name]></[namespace]-[name]>
+   ```
```
Comment (blackfalcon) [inline template/docs/topics/install.md:19]: This never works out as you hope it does with markdown. I'd suggest removing the numbers and just use headers.

File: template/package.json [hunk 6]
```diff
@@ -133,11 +133,12 @@
     "web components"
   ],
   "scripts": {
-    "build": "npm-run-all build:sass sass:render dist:js build:api test bundler postinstall",
+    "build": "npm-run-all build:sass sass:render dist:js test build:api build:markdownDocs bundler postinstall",
     "build:ci": "npm-run-all sweep build",
     "build:api": "wca analyze 'src/[namespace]-[name].js' --outFiles docs/api.md",
     "build:demo": "npm-run-all build demo:rm:build demo:new:build demo:copy:index demo:copy:demo demo:update:index",
     "build:dev:assets": "npm-run-all build:sass:demo build:sass:component postCss:component sass:render",
+    "build:markdownDocs": "node util/generateDocs.js",
```
Comment (blackfalcon) [inline template/package.json:141]:
```suggestion
    "build:markdownDocs": "node scripts/generateDocs.js",
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: template/docs/topics/development.md (hunk 4 above)
Comment (jason-capsule42) [reply template/docs/topics/development.md:1]: As we discussed last week I plan for this to be exposed on the doc site. It's also ideal for scoped education reference.

File: template/docs/topics/componentInstall.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
+# Install [namespace]-[name]
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply template/docs/topics/componentInstall.md:1]: As we discussed last week I plan for this to be exposed on the doc site. It's also ideal for scoped education reference.

File: template/docs/topics/demoDeployment.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+To deploy a demo version of the component for review, run `npm run build:demo` to create a `./build` directory that can be pushed to any static server.
+
+<small>Built from WC-Generator v[genVersion]</small>
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply template/docs/topics/demoDeployment.md:3]: I assume you mean line 3?

File: template/docs/topics/componentInstall.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (blackfalcon) [reply template/docs/topics/componentInstall.md:1]: With each component section of the docsite, the `install` page points to the README file. How is this different and why the need for the change? 

This really should be part of a larger content architecture discussion that is part of the docsite redesign efforts.

File: template/docs/topics/development.md (hunk 4 above)
Comment (blackfalcon) [reply template/docs/topics/development.md:1]: Same reply from `componentInstall.md`

**Comment by blackfalcon** [comment]
I am assuming that this has been abandoned in favor for https://github.com/AlaskaAirlines/WC-Generator/pull/292

**Pull request closed without merging**